### ISSUE TITLE
no resources created under rook-ceph namespace

### ISSUE THREAD

**Issue by delubi** [rook/rook#2087]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

I'm new to rook. Trying to create a rook cluster but not able to get resources created under rook-ceph namespace.

Below were the steps performed.

kubectl --namespace kube-system create sa tiller
kubectl create clusterrolebinding tiller --clusterrole cluster-admin --serviceaccount=kube-system:tiller
kubectl --namespace kube-system patch deploy/tiller-deploy -p '{"spec": {"template": {"spec": {"serviceAccountName": "tiller"}}}}'

helm repo add rook-beta https://charts.rook.io/beta
helm install --namespace rook-ceph-system rook-beta/rook-ceph

[root@k8smaster rook]# cat cluster.yaml
apiVersion: v1
kind: Namespace
metadata:
  name: rook-ceph
---
apiVersion: ceph.rook.io/v1beta1
kind: Cluster
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  dataDirHostPath: /var/lib/rook
  dashboard:
    enabled: true
  storage:
    useAllNodes: true
    useAllDevices: false
    config:
      databaseSizeMB: "4096"
      journalSizeMB: "1024"

kubectl create -f cluster.yaml

[root@k8smaster etc]# kubectl -n rook-ceph get pod
No resources found.

[root@k8smaster etc]# kubectl -n rook-ceph-system get pod
NAME                                  READY     STATUS    RESTARTS   AGE
rook-ceph-agent-qm82w                 1/1       Running   0          50m
rook-ceph-operator-5457bb645b-z454n   1/1       Running   0          51m
rook-discover-kbjmf                   1/1       Running   0          50m

**Expected behavior:**

Expect to see pod running under rook-ceph namespace.

**How to reproduce it (minimal and precise):**

**Environment**:
* OS (e.g. from /etc/os-release): 

[root@k8smaster etc]# cat centos-release
CentOS Linux release 7.5.1804 (Core)

* Kernel (e.g. `uname -a`): 

[root@k8smaster etc]# uname -a
Linux k8smaster 3.10.0-862.3.2.el7.x86_64 #1 SMP Mon May 21 23:36:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

* Cloud provider or hardware configuration:

VM running in VMware vCenter

* Rook version (use `rook version` inside of a Rook Pod):

N/A. Not able to get rook pod created

* Kubernetes version (use `kubectl version`):

[root@k8smaster etc]# kubectl version
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:13:03Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:00:59Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):

Created on a single node with kubeadm.

* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)):

N/A. Not able to get rook pod creaetd.

**Comment by galexrt** [comment]
@delubi Could you please add code tags to your comment. I'm unable to see how the indent in the cluster.yaml is you posted.

**Comment by delubi** [comment]
@galexrt Please see the content of cluster.yaml as below. Thanks!
`
apiVersion: v1
kind: Namespace
metadata:
  name: rook-ceph
---
apiVersion: ceph.rook.io/v1beta1
kind: Cluster
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  dataDirHostPath: /var/lib/rook
  dashboard:
    enabled: true
  storage:
    useAllNodes: true
    useAllDevices: false
    config:
      databaseSizeMB: "4096"
      journalSizeMB: "1024"
`

**Comment by galexrt** [comment]
@delubi You need to write it like that:
\```
CODE
\```

Please edit your comment with those code blocks so the indent is visible on each line.

**Comment by delubi** [comment]
[cluster.zip](https://github.com/rook/rook/files/2357587/cluster.zip)

@galexrt Let me try again as below. Also attached is the zip file that contains the cluster.yaml file.

```
apiVersion: v1
kind: Namespace
metadata:
  name: rook-ceph
---
apiVersion: ceph.rook.io/v1beta1
kind: Cluster
metadata:
  name: rook-ceph
  namespace: rook-ceph
spec:
  dataDirHostPath: /var/lib/rook
  dashboard:
    enabled: true
  storage:
    useAllNodes: true
    useAllDevices: false
    config:
      databaseSizeMB: "4096"
      journalSizeMB: "1024"
```

**Comment by galexrt** [comment]
@delubi Try to create the RBAC definitions from here https://rook.io/docs/rook/v0.8/ceph-cluster-crd.html#common-cluster-resources and check that after a Rook Ceph operator restart (delete the Pod) the Cluster is successfully starting. If not, please let us know and attach the Rook Ceph operator logs.

**Comment by guidoilbaldo** [comment]
Hi, I have the same problem.
Running minikube v0.28.2 on MacOS (freshly installed) and kubectl version `1.11`.
I'm able to start operator, agent and discovery pods, then I simply try to start the cluster with the example posted here -> https://rook.github.io/docs/rook/master/ceph-quickstart.html
with `kubectl create -f cluster.yaml`.

```
~ kubectl create -f cluster-resources.yaml                                                                                                                                                           12:50:37
namespace/rook-ceph created
serviceaccount/rook-ceph-cluster created
role.rbac.authorization.k8s.io/rook-ceph-cluster created
rolebinding.rbac.authorization.k8s.io/rook-ceph-cluster-mgmt created
rolebinding.rbac.authorization.k8s.io/rook-ceph-cluster created
```

but then `no resources found` keeps popping up.
Any ideas?

**Comment by delubi** [comment]
@galexrt Looks good now. Thank you!

[root@k8smaster rook]# kubectl get pod -n rook-ceph
NAME                                    READY     STATUS      RESTARTS   AGE
rook-ceph-mgr-a-88d56c679-w5pzt         1/1       Running     0          14m
rook-ceph-mon0-5xc7t                    1/1       Running     0          15m
rook-ceph-mon1-xbp4b                    1/1       Running     0          15m
rook-ceph-mon2-hmqw6                    1/1       Running     0          15m
rook-ceph-osd-id-0-85645d66d8-nkcdh     1/1       Running     0          13m
rook-ceph-osd-prepare-k8smaster-xfbc8   0/1       Completed   0          13m

**Comment by galexrt** [comment]
Closing as this issue has been resolved.
